diff --git .eslintrc.cjs .eslintrc.cjs
index 1d0a7fa..22524e5 100644
--- .eslintrc.cjs
+++ .eslintrc.cjs
@@ -3,7 +3,7 @@ require('@rushstack/eslint-patch/modern-module-resolution')
 
 module.exports = {
   root: true,
-  'ignorePatterns': ['*.config.js'],
+  'ignorePatterns': ['*.config.js', '!.storybook'],
   'extends': ['plugin:vue/vue3-essential', 'eslint:recommended', '@vue/eslint-config-typescript', '@vue/eslint-config-prettier/skip-formatting', 'plugin:storybook/recommended'],
   parserOptions: {
     ecmaVersion: 'latest'
